### PR TITLE
[IR] introduced additional field 'inlineFunctionSymbol' to IrReturnableBlock

### PR DESCRIPTION
**Pull request by vvlevchenko**
This change let backend calculate and bind right function to the code block on generation debug information
e.g. for Kotlin/Native backend this let codegnerator to produce DILocation(..., inlinedAt:..) with correct value
for inlinedAt parameter, thus signifiacntly improve behaviour of debugger.

https://github.com/JetBrains/kotlin-native/pull/2849

### CODE REVIEW

File: compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/expressions/impl/IrBlockImpl.kt [hunk 1]
```diff
@@ -65,13 +65,14 @@ fun IrBlockImpl.inlineStatement(statement: IrStatement) {
...
 }
 
-class IrReturnableBlockImpl(
-    startOffset: Int,
-    endOffset: Int,
-    type: IrType,
-    override val symbol: IrReturnableBlockSymbol,
-    origin: IrStatementOrigin? = null,
-    override val sourceFileSymbol: IrFileSymbol? = null
+open class IrReturnableBlockImpl(
```
Comment (romanart) [inline compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/expressions/impl/IrBlockImpl.kt]: Why is it `open`?

Comment (vvlevchenko) [reply compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/expressions/impl/IrBlockImpl.kt]: Ooops.

File: compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/expressions/IrBlock.kt [hunk 2]
```diff
@@ -38,6 +41,13 @@ interface IrComposite : IrContainerExpression {
...
-}
\ No newline at end of file
+    val inlineFunctionSymbol: IrFunctionSymbol?
+}
+
+val IrReturnableBlock.sourceFileSymbol: IrFileSymbol?
+    get() = inlineFunctionSymbol?.owner?.file?.symbol
+
+
+val IrReturnableBlock.sourceFileName: String
```
Comment (bashor) [inline compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/expressions/IrBlock.kt]: how about deprecating it?

Comment (vvlevchenko) [reply compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/expressions/IrBlock.kt]: ok

**Pull request merged**